shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1639,22 +1626,37 @@ - (CGRect)firstRectForRange:(UITextRange*)range {
 }
 
 - (CGRect)caretRectForPosition:(UITextPosition*)position {
-  // TODO(cbracken) Implement.
-
-  // As of iOS 14.4, this call is used by iOS's
-  // _UIKeyboardTextSelectionController to determine the position
-  // of the floating cursor when the user force touches the space
-  // bar to initiate floating cursor.
-  //
-  // It is recommended to return a value that's roughly the
-  // center of kSpacePanBounds to make sure the floating cursor
-  // has ample space in all directions and does not hit kSpacePanBounds.
-  // See the comments in beginFloatingCursorAtPoint.
-  return CGRectZero;
-}
-
-- (CGRect)bounds {
-  return _isFloatingCursorActive ? kSpacePanBounds : super.bounds;
+  NSInteger index = ((FlutterTextPosition*)position).index;
+  // Get the bounds of the characters before and after the requested caret position.
+  NSArray<UITextSelectionRect*>* rects =
+      [self selectionRectsForRange:[FlutterTextRange
+                                       rangeWithNSRange:fml::RangeForCharactersInRange(
+                                                            self.text,
+                                                            NSMakeRange(MAX(0, index - 1), 1))]];
+  if (rects.count == 0) {
+    return CGRectZero;
+  }
+  if (index == 0) {
+    // There is no character before the caret, so this will be the bounds of the character after the
+    // caret position.
+    CGRect characterAfterCaret = rects[0].rect;
+    // Return a zero-width rectangle 30% in from the left edge of the character after the caret
```

Comment:
Does this work RTL?

Reply:
I thought it did but I tested the wrong way, looking into changes now to handle proper RTL unicode chars.

Reply:
I think there is a bug in Skia getGlyphPositionAtCoordinate in RTL, looking at that now